src/ripple/rpc/handlers/GetAggregatePrice.cpp

```diff
@@ -0,0 +1,329 @@
+    // for the Oracle is not found in the inner loop
+    STObject const* prevChain = nullptr;
+
+    Meta meta = nullptr;
+    while (true)
+    {
+        if (prevChain == chain)
+            return;
+
+        if (!oracle || f(*oracle) || isNew)
+            return;
+
+        if (++history > maxHistory)
+            return;
+
+        uint256 prevTx = chain->getFieldH256(sfPreviousTxnID);
+        std::uint32_t prevSeq = chain->getFieldU32(sfPreviousTxnLgrSeq);
+
+        auto const ledger = context.ledgerMaster.getLedgerBySeq(prevSeq);
+        if (!ledger)
+            return;
+
+        meta = ledger->txRead(prevTx).second;
+
+        for (STObject const& node : meta->getFieldArray(sfAffectedNodes))
+        {
+            if (node.getFieldU16(sfLedgerEntryType) != ltORACLE)
+            {
+                continue;
+            }
+
+            prevChain = chain;
+            chain = &node;
+            isNew = node.isFieldPresent(sfNewFields);
+            // if a meta is for the new and this is the first
+            // look-up then it's the meta for the tx that
+            // created the current object; i.e. there is no
+            // historical data
+            if (isNew && history == 1)
+                return;
+
+            oracle = isNew
+                ? &static_cast<const STObject&>(node.peekAtField(sfNewFields))
+                : &static_cast<const STObject&>(
+                      node.peekAtField(sfFinalFields));
+            break;
+        }
+    }
+}
+
+// Return avg, sd, data set size
+static std::tuple<STAmount, Number, std::uint16_t>
+getStats(
+    Prices::right_const_iterator const& begin,
+    Prices::right_const_iterator const& end)
+{
+    STAmount avg{noIssue(), 0, 0};
+    Number sd{0};
+    std::uint16_t size = std::distance(begin, end);
+    avg = std::accumulate(
+        begin, end, avg, [&](STAmount const& acc, auto const& it) {
+            return acc + it.first;
+        });
+    avg = divide(avg, STAmount{noIssue(), size, 0}, noIssue());
+    if (size > 1)
+    {
+        sd = std::accumulate(
+            begin, end, sd, [&](Number const& acc, auto const& it) {
+                return acc + (it.first - avg) * (it.first - avg);
+            });
+        sd = root2(sd / (size - 1));
+    }
+    return {avg, sd, size};
+};
+
+/**
+ * oracles: array of {account, oracle_sequence}
+ * base_asset: is the asset to be priced
+ * quote_asset: is the denomination in which the prices are expressed
+ * trim : percentage of outliers to trim [optional]
+ * time_threshold : defines a range of prices to include based on the timestamp
+ *   range - {most recent, most recent - time_threshold} [optional]
+ */
+Json::Value
+doGetAggregatePrice(RPC::JsonContext& context)
+{
+    auto const& params(context.params);
+    std::shared_ptr<ReadView const> ledger;
+    auto result = RPC::lookupLedger(ledger, context);
+    if (!ledger)
+        return result;
+
+    constexpr std::uint16_t maxOracles = 200;
```

Comment:
Why such a high limit? Couldn't that be tough for performance?

Reply:
It is sort of arbitrary. But I figured if we could delete up to 1000 objects on account delete then processing 200 oracles should be ok. Not sure if it's a fair comparison.

Reply:
Account delete costs 2 XRP, an RPC is free. That was my thought.

Reply:
Makes sense. But we could also have API's that, for instance return substantial amount of data. `ledger_data` comes to mind with a limit of 2048 objects. Iteration/allocation could be a performance hit, no?

Reply:
Yeah, that's fair. Just making sure it's been considered. This is probably a question for @sophiax851.